.github/workflows/build.yml

```diff
@@ -0,0 +1,108 @@
+name: Build
+
+on: 
+  push:
+    branches:
+      - 'master'
+  pull_request:
+
+
+jobs:
+  Build:
+    name: Build appimagetool
+    runs-on: ubuntu-20.04
+    strategy:
+      matrix:
+        include:
+          - ARCH: 'x86_64'
+            docker-container: 'appimagebuild'
+          - ARCH: 'i386'
+            docker-container: 'appimagebuild-i386'
+          - ARCH: 'armhf'
+            docker-container: 'appimagebuild-armhf-cross'
+          - ARCH: 'aarch64'
+            docker-container: 'appimagebuild-aarch64-cross'
+      fail-fast: false
+    env:
+      DOCKER_IMAGE: quay.io/appimage/${{ matrix.docker-container }}
+      ARCH: ${{ matrix.ARCH }}
+
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        submodules: true
+
+    - name: Install dependencies for ARM and AARCH64
+      if: endsWith(matrix.docker-container, 'cross') 
+      run: |
+        sudo apt-get update
+        sudo apt-get -y install binfmt-support qemu-user-static
+
+    - name: Build 
+      shell: bash
+      run: |
+        if [[ "$ARCH" == "i386" ]]; then
+          sudo dpkg --add-architecture i386
+          sudo apt-get update
+        fi
+        bash -ex travis/travis-build.sh
+
+    - name: List files
+      run: |
+        ls -lh build/out/*
+
+    - name: Upload artifact
+      uses: actions/upload-artifact@v1.0.0
+      with:
+        # Artifact name
+        name: appimagetool-${{ matrix.ARCH }}.AppImage.zip
+        path: build/out
+
+
+  Release:
+    needs: [Build]
+    runs-on: ubuntu-latest
+ 
+    steps:
+    - uses: actions/download-artifact@v1
+      with:
+        name: appimagetool-x86_64.AppImage.zip
```

Comment:
`upload-action`  by default uploads the file as a `zip` archive, without extension. I suppose, it would be more helpful if it was uploaded and downloaded with the `zip` extension. GitHub internal artifactory can only hold zips. A user downloading from build artifacts (ideally, they shouldn't ) would receive stuff which are zips, normally without an extension. Adding a `.zip` extension would ideally let the user know what tool he/she must use to open the archive. Lets not let them mistake the zip for the appimage. and downloading the artifact would extract the zip as a directory. So, we are back..